src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder string = new StringBuilder();
+        string.append("Report for period ").append(dateFrom).append(" - ")
+                .append(dateTo);
+        for (int i = 0; i < names.length; i++) {
+            int money = 0;
+            for (String eachData : data) {
+                String[] eachDataArray = eachData.split(" ");
+                if (names[i].equals(eachDataArray[1])
+                        && (getDate(eachDataArray[0]).equals(getDate(dateFrom))
+                        || getDate(eachDataArray[0]).after(getDate(dateFrom)))
+                        && (getDate(eachDataArray[0]).equals(getDate(dateTo))
+                        || getDate(eachDataArray[0]).before(getDate(dateTo)))) {
+                    money += Integer.parseInt(eachDataArray[2])
+                            * Integer.parseInt(eachDataArray[3]);
```

Comment:
move indices to constants because they look like magic numbers

consider using local variable for `getDate(eachDataArray[0])`